packages/shared/lib/wallet.ts

```diff
@@ -586,6 +586,28 @@ export const asyncCreateAccount = (): Promise<void> =>
         )
     })
 
+export const asyncRemoveWalletAccount = (accountId: string): Promise<void> =>
+    new Promise<void>((resolve, reject) => {
+        api.removeAccount(accountId, {
+            onSuccess() {
+                /**
+                 * CAUTION: If an account is successfully removed in wallet.rs then it should also be
+                 * removed in the Firefly store. This is "inefficient" (esp. for batch deletes) but it
+                 * at least ensures data integrity / consistency between Firefly and the backend.
+                 */
+                get(wallet).accounts.update((_accounts) => _accounts.filter((wa) => wa.id !== accountId))
```

Comment:
I think we should avoid shortening variable names

Reply:
I think that findIndex and splice would be faster than filter, but it would probably be marginal.

Reply:
> I think we should avoid shortening variable names

I generally agree, although I personally think it's fine using it in functional methods like `map` or `filter`, unless the operation is a bit more complex / requires a few lines to write. 

> I think that findIndex and splice would be faster than filter, but it would probably be marginal.

It probably would, although I tend to stick to the functional style rather than imperative. You're right though, it doesn't really matter since both have upper bounds of `O(n)`, assuming `splice` is constant time.